### ISSUE TITLE
error reading llm response：An existing connection was forcibly closed by the remote host.

### ISSUE THREAD

**Issue by yaosd99** [ollama/ollama#6937]
### What is the issue?

![bug-help](https://github.com/user-attachments/assets/07b1633d-7fe1-49c6-8267-d0432331f143)

***Cannot import images.
Please~~

### OS

Windows

### GPU

Nvidia

### CPU

AMD

### Ollama version

0.3.11

**Comment by srikary12** [comment]
I'll take a look at it.

**Comment by yaosd99** [comment]
> This model loads correctly for me on an nvidia GPU on windows. Can you share your server log so we can see what may be going wrong?
> 
> https://github.com/ollama/ollama/blob/main/docs/troubleshooting.md

This is a reproduced log file, please refer to it~
[server.log](https://github.com/user-attachments/files/17128701/server.log)

**Comment by srikary12** [comment]
The logs look clear and model loads correctly for me. Can you restart your ollama and check of the issue still persists?

**Comment by yaosd99** [comment]
> 日志看起来很清晰，对我来说模型加载正确。您能否重新启动 ollama 并检查问题仍然存在？

Yes, I tried various methods including restarting and rebuilding the environment, but still encountered the aforementioned issues~~~~~~~
=。=

**Comment by dhiltgen** [comment]
The attached logs don't seem to show a crash.  I actually see to `200` responses after the load, implying things worked correctly.  @yaosd99 can you clarify?  Was this log capturing the exchange where the client received the error `error reading llm response：An existing connection was forcibly closed by the remote host.`

**Comment by yaosd99** [comment]
> The attached logs don't seem to show a crash. I actually see two responses after the load, implying things worked correctly. @yaosd99 can you clarify? Was this log capturing the exchange where the client received the error `200``error reading llm response：An existing connection was forcibly closed by the remote host.`

After testing, the server. log does not receive the error displayed by cmd, and the 5-digit number of "xxxxx" in the error "127.0.0.1: xxxxx" changes every time

**Comment by dhiltgen** [comment]
Ollama spawns a child process on a random port number for every model load.  I've tried a few different approaches but haven't managed to reproduce this crash, so I'm still not sure why it crashes on your system.  Setting OLLAMA_DEBUG=1 might yield some more logging that's helpful.  You could also try to force it back to use the CUDA v11 runner instead of v12 by setting OLLAMA_LLM_LIBRARY=cuda_v11

Do other models work correctly?

**Comment by steveseguin** [comment]
I'm having the same error. It's probably just me being stupid, but things work if I don't include an image.

Windows 11. Titan RTX. Ollama 0.4.0-rc5 using Llama 3.2-vision:latest

8-KB square image uploaded, with a message "describe".  Fails with the same message as the OP.

![image](https://github.com/user-attachments/assets/962f5eb5-64d9-40d2-afef-6254e6134750)
![image](https://github.com/user-attachments/assets/25b0a6eb-5e32-4b22-9079-0dd222ca08ce)
![image](https://github.com/user-attachments/assets/ef674d86-1131-4e37-932a-0d4649b9cbdd)

![image](https://github.com/user-attachments/assets/eb5a7865-d72c-4699-b198-31e9efb8e66f)
![image](https://github.com/user-attachments/assets/6431f9f3-3691-4949-be01-e2bf6578335d)


server.log shows this:
```
Exception 0xc0000005 0x0 0x241ea9da000 0x7ff97aff2edb
PC=0x7ff97aff2edb
signal arrived during external code execution
```
[server.log](https://github.com/user-attachments/files/17517383/serverlog.log)

Hope it helps.

**Comment by jessegross** [comment]
@steveseguin That is a different issue - yours is the same as #7362 

That one has been fixed in current `main` if you build from source or in the next RC.

**Comment by yaosd99** [comment]
> Ollama spawns a child process on a random port number for every model load. I've tried a few different approaches but haven't managed to reproduce this crash, so I'm still not sure why it crashes on your system. Setting OLLAMA_DEBUG=1 might yield some more logging that's helpful. You could also try to force it back to use the CUDA v11 runner instead of v12 by setting OLLAMA_LLM_LIBRARY=cuda_v11
> 
> Do other models work correctly?

After my testing, the previous bug was due to the model's inability to recognize a portion of the images, which prevented it from returning the inference results to Ollama. It's not Ollama's problem, it's the problem with the LLM.

**Comment by HappyManRus** [comment]
I have the same problem only in llm "deepseek-coder-v2."
`'error reading llm response: read tcp 127.0.0.1:59935->127.0.0.1:59933: wsarecv: An existing connection was forcibly closed by the remote host.'`

If I send a small query, work perfect. But when I query like "make unit test for PHP code....", the system 4-5 minutes, and then I see this error.

Other models work correctly. I tried more than ten.

![image](https://github.com/user-attachments/assets/4d292613-18ba-47e5-830c-9f9d7d9e67d1)

**Comment by dhiltgen** [comment]
@HappyManRus can you check your server log?  There are a number of existing issues tracking deepseek crashes.  Depending on what the crash was, there may be a workaround you can try until we get it fixed.

**Comment by HappyManRus** [comment]
> @HappyManRus can you check your server log? There are a number of existing issues tracking deepseek crashes. Depending on what the crash was, there may be a workaround you can try until we get it fixed.

Hello. 
[@dhiltgen](https://github.com/dhiltgen), I hope it helps.

[server_copy.log](https://github.com/user-attachments/files/17820237/server_copy.log)

**Comment by jessegross** [comment]
@HappyManRus Your issue is the same as #5975 - there is a workaround there.

I'm going to go ahead and close this issue as it seems that the original issue has been fixed. For other issues, please follow up in the linked bugs if needed.
